viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -283,7 +286,9 @@ export class MeasurementTool extends Cognite3DViewerToolBase {
    * @param enable
    */
   setMeasurementLabelsVisible(enable: boolean): void {
-    this._htmlOverlay.visible(enable);
+    if (this._showMeasurements) {
+      this._htmlOverlay.visible(enable);
+    }
```

Comment:
This seems a bit weird.. Can't the visibility of the labels be toggled even if the lines are invisible? For instance, if you hide all lines in the example, then toggle labels off, then show the lines again, the labels will also be visible, even though their checkbox is not checked.

Reply:
When measurement is in hide state, all objects related to measurement that is line, labels should be hidden and I think even if we try to enable labels in measurement hidden state, it should not change the labels visibility